dnsapi/dns_world4you.sh

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env sh
+
+# World4You - www.world4you.com
+# Lorenz Stechauner, 2020 - https://www.github.com/NerLOR
+
+WORLD4YOU_API="https://my.world4you.com/en"
+PAKETNR=''
+TLD=''
+RECORD=''
+
+################ Public functions ################
+
+# Usage: dns_world4you_add <fqdn> <value>
+dns_world4you_add() {
+  fqdn="$1"
+  value="$2"
+  _info "Using world4you"
+  _debug fulldomain "$fqdn"
+  _debug txtvalue "$value"
+
+  _login
+  if [ "$?" != 0 ]; then
+    return 1
+  fi
+
+  export _H1="Cookie: W4YSESSID=$sessid"
+  form=$(_get "$WORLD4YOU_API/dashboard/paketuebersicht")
+  _get_paketnr "$fqdn" "$form"
+  paketnr="$PAKETNR"
+  tld="$TLD"
+  record="$RECORD"
+  if [ -z "$paketnr" ]; then
+    _err "Unable to parse paketnr"
+    return 3
+  fi
+  _debug paketnr "$paketnr"
+
+  export _H1="Cookie: W4YSESSID=$sessid"
+  form=$(_get "$WORLD4YOU_API/$paketnr/dns")
+  formiddp=$(echo "$form" | grep 'AddDnsRecordForm\[uniqueFormIdDP\]' | sed 's/^.*name="AddDnsRecordForm\[uniqueFormIdDP\]" value="\([^"]*\)".*$/\1/')
+  formidttl=$(echo "$form" | grep 'AddDnsRecordForm\[uniqueFormIdTTL\]' | sed 's/^.*name="AddDnsRecordForm\[uniqueFormIdTTL\]" value="\([^"]*\)".*$/\1/')
+  form_token=$(echo "$form" | grep 'AddDnsRecordForm\[_token\]' | sed 's/^.*name="AddDnsRecordForm\[_token\]" value="\([^"]*\)".*$/\1/')
+  if [ -z "$formiddp" ]; then
+    _err "Unable to parse form"
+    return 3
+  fi
+
+  _ORIG_ACME_CURL="$_ACME_CURL"
+  _ACME_CURL=$(echo "$_ACME_CURL" | sed 's/ -L / /')
```

Comment:
why do you need to change the curl here?
Don't assume it's using curl,  it can be using with `wget` or any othe http tool in future.

Reply:
The `-L` option lets curl follow redirects, as you know. The problem is, that the response I get is a `302` redirect to the same page. Usually a browser follows this redirect with `GET`. Curl does not, it keeps the `POST`. This results in an infinite loop, until (after 50 redirects) curl exits with an error. 

I did not find any function parameter or global variable to let curl/wget NOT follow redirects, so I had to write this ugly workaround. Any help to make this a bit cleaner would be appreciated :)

Reply:
please rebase code.
you can use this code before requests:

```sh
ACME_HTTP_NO_REDIRECTS=1
_resethttp

```

Reply:
The code snippet is basically working, with a few modifications:
```
_resethttp
export ACME_HTTP_NO_REDIRECTS=1
```
(`_resethttp` resets `ACME_HTTP_NO_REDIRECTS` and ShellCheck wants `export` before the variable)